book/src/storage/transfer-to-object.md

```diff
@@ -1,4 +1,132 @@
-# Transfer to Object?
+# Transfer to Object
 
-The `transfer::transfer` call takes the receiver `address` as the second argument, and while in most
-of the cases it is an account address, it can also be an address of an object. In this case,
+Previously, we have explained how [transfer](./storage-functions.md#transfer) works in relation to
+accounts. However, there is an additional behaviour allowed by the
+[Sui Object Model](./../object/object-model.md) - transferring objects to other objects. If certain
+conditions are met (which we explain in this section), objects can be sent to and _received_ from
+other objects.
+
+<div class="warning">
+
+Warning: recipient object must be implemented in a way that allows receiving. Attempt to send an
+object to an object that does not implement receiving can result in asset loss.
+
+</div>
+
+## Background
+
+Every object in Sui has its own `UID` which is represented by the
+[address type](./../move-basics/address.md). This address can be used to query the object's data,
+and to perform account queries, such as getting the list of owned objects. This property of the
+system lays the foundation for the transfer to object feature.
+
+> In this section, by _object address_ we mean the address value stored in the `UID` of the object.
+
+## Transfer to Object
+
+An object can be transferred to another object by calling the `transfer` function, or its public
+version - `public_transfer`. The behaviour is identical to the transfer to account.
+
+```move
+// File: sui-framework/sources/transfer.move
+module sui::transfer;
+
+public fun transfer<T: key>(object: T, recipient: address) {}
+public fun public_transfer<T: key + store>(object: T, recipient: address) {}
+```
+
+## Receiving Objects
+
+While objects _owned by accounts_ can be used in the transaction directly (given that the sender of
+the transaction is the owner), objects transferred to other objects first need to be _received_
+through their owner. The owner object needs to implement a custom function that will call the
+`receive` function from the `transfer` module.
+
+```move
+// File: sui-framework/sources/transfer.move
+module sui::transfer;
+
+/// Special type which cannot be constructed in Move but can be passed as a
+/// special input to a PTB.
+public struct Receiving<phantom T: key> has drop {
+    id: ID,
+    version: u64,
+}
+
+/// Private receiving function for `key`-only objects.
+public fun receive<T: key>(parent: &mut UID, to_receive: Receiving<T>): T { /* ... */ }
+
+/// Public receiving function for `key + store` objects.
+public fun public_receive<T: key + store>(parent: &mut UID, to_receive: Receiving<T>): {
+  /* ... */
+}
+```
+
+The `Receiving` type is a special type that references an object-owned object. It cannot be
```

Comment:
I think including this part in the example would be very helpful - I don't yet see how the end-to-end process of transferring works. e.g. if Alice wants to transfer an object to a PostOffice owned by Bob, does Bob have to execute a PTB that constructs a `Receiving` and hands it to the `PostOffice`'s `receive` function?